src/syscheckd/src/create_db.c

```diff
@@ -44,129 +44,281 @@ static const char *FIM_EVENT_MODE[] = {
-            return;
+            return; // LCOV_EXCL_LINE
         }
     // In case of a deletion, DBSync is not going to return an array.
     } else {
         dbsync_event = result_json;
     }
 
-    if (json_path = cJSON_GetObjectItem(dbsync_event, FILE_PRIMARY_KEY), json_path == NULL) {
+    // In case of deletions, latest_entry is NULL, so we need to get the path from the json event
+    if (resultType == DELETED) {
+        cJSON *path_cjson = cJSON_GetObjectItem(dbsync_event, "path");
+        if (path_cjson == NULL) {
+            goto end;
+        }
+        path = cJSON_GetStringValue(path_cjson);
+    } else if (txn_context->latest_entry != NULL) {
+        path = txn_context->latest_entry->file_entry.path;
+    } else {
         goto end;
     }
 
-    path = cJSON_GetStringValue(json_path);
-
     if (configuration = fim_configuration_directory(path), configuration == NULL) {
         goto end;
     }
 
-    if (configuration->options & CHECK_SEECHANGES) {
+    if (configuration->options & CHECK_SEECHANGES && resultType != DELETED) {
         diff = fim_file_diff(path, configuration);
     }
 
     switch (resultType) {
         case INSERTED:
-            event_data->evt_data->type = FIM_ADD;
+            txn_context->evt_data->type = FIM_ADD;
             break;
 
         case MODIFIED:
-            event_data->evt_data->type = FIM_MODIFICATION;
+            txn_context->evt_data->type = FIM_MODIFICATION;
             break;
 
         case DELETED:
             if (configuration->options & CHECK_SEECHANGES) {
                 fim_diff_process_delete_file(path);
             }
-            event_data->evt_data->type = FIM_DELETE;
-            event_data->db_full = false;
+
+            txn_context->evt_data->type = FIM_DELETE;
+            txn_context->db_full = false;
 
             break;
 
         case MAX_ROWS:
-            event_data->db_full = true;
+            txn_context->db_full = true;
             mdebug1("Couldn't insert '%s' entry into DB. The DB is full, please check your configuration.", path);
 
         // Fallthrough
         default:
             goto end;
     }
 
-    json_event = fim_dbsync_json_event(path, diff, dbsync_event, configuration, event_data->evt_data);
+    json_event = cJSON_CreateObject();
+    data = cJSON_CreateObject();
 
-    if (json_event != NULL) {
-        send_syscheck_msg(json_event);
+    if (json_event == NULL || data == NULL) {
+        goto end; // LCOV_EXCL_LINE
     }
 
+    cJSON_AddStringToObject(json_event, "type", "event");
+    cJSON_AddItemToObject(json_event, "data", data);
+
+    cJSON_AddStringToObject(data, "path", path);
+    cJSON_AddNumberToObject(data, "version", 2.0);
+    cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[txn_context->evt_data->mode]);
+    cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[txn_context->evt_data->type]);
+    //cJSON_AddNumberToObject(data, "timestamp", time(NULL)); This should be the last_event field
```

Comment:
Let's clean the commented code

Reply:
This is just to avoid a false positive when the entries are updated because this field is changed in every syncRow operation. This will be fixed in https://github.com/wazuh/wazuh/issues/12166.

I think we should leave this and uncomment this line when this issue is completed.